pkg/Cpanel/Security/Advisor/Assessors/Trojans.pm

```diff
@@ -73,7 +92,521 @@ sub _check_for_libkeyutils {
+            if ( -l $proc_pid_exe && readlink($proc_pid_exe) =~ m{ \(deleted\) }xms ) {
+                next if ( ( stat($proc_pid_exe) )[7] > $max_bin_size );
+                push @apache_bins, $proc_pid_exe;
+            }
+        }
+    }
+
+    for my $check_bin (@apache_bins) {
+        my $httpd;
+        if ( open my $fh, '<', $check_bin ) {
+            local $/;
+            $httpd = <$fh>;
+            close $fh;
+        }
+
+        next if !$httpd;
+
+        if ( $httpd =~ /(open_tty|hangout|ptsname|Qkkbal)/ ) {
+            $signature   = $check_bin . ": \"" . $1 . "\"";
+            $has_cdorked = 1;
+            last;
+        }
+    }
+    if ( $has_cdorked == 1 ) {
+        $self->add_bad_advice(
+            'text'       => [ "CDORKED rootkit check: Evidence of the CDORKED A rootkit was found." ],
+            'suggestion' => [
+                'Check the following for more information "[output,url,_1,We Live Security More Information,_2,_3]"',
+                'http://www.welivesecurity.com/wp-content/uploads/2014/03/operation_windigo.pdf',
+                'target',
+                '_blank'
+            ],
+        );
+    }
+}
+
+sub _check_for_cdorked_B {
+    my ($self) = @_;
+    my $has_cdorked_b = 0;
+    my @files = ( '/usr/sbin/arpd ', '/usr/sbin/tunelp ', '/usr/bin/s2p ' );
+    my $cdorked_files;
+
+    for my $file (@files) {
+        if ( -e $file ) {
+            $has_cdorked_b = 1;
+            $cdorked_files .= "[$file] ";
+        }
+    }
+
+    if ( $has_cdorked_b == 1 ) {
+        $self->add_bad_advice(
+            'text'       => [ "CDORKED rootkit check: The following file(s) [_1] was/were found, indicating the possibility of the CDORKED B rootkit.", $cdorked_files ],
+            'suggestion' => [
+                'Check the following for more information "[output,url,_1,We Live Security More Information,_2,_3]"',
+                'http://www.welivesecurity.com/2013/04/26/linuxcdorked-new-apache-backdoor-in-the-wild-serves-blackhole/',
+                'target',
+                '_blank'
+            ],
+        );
+    }
+}
+
+sub _check_for_libkeyutils_filenames {
+    my ($self) = @_;
+    my $bad_libs;
+    my @dirs  = qw( /lib /lib64 );
+    my @files = qw(
+      libkeyutils.so.1.9
+      libkeyutils-1.2.so.0
+      libkeyutils-1.2.so.2
+      libkeyutils.so.1.3.0
+      libkeyutils.so.1.3.2
+      libns2.so
+      libns5.so
+      libpw3.so
+      tls/libkeyutils.so.1
+      tls/libkeyutils.so.1.5
+    );
+
+    for my $dir (@dirs) {
+        next if !-e $dir;
+        for my $file (@files) {
+            if ( -f "${dir}/${file}" and !-z "${dir}/${file}" ) {
+                $bad_libs .= "\t${dir}/${file}\n";
+            }
+        }
+    }
+
+    if ($bad_libs) {
+        $self->add_bad_advice(
+            'text'       => [ "Libkey rootkit check: The following suspicious system libraries [_1] were found which could indicate a root level compromise.", $bad_libs ],
```

Comment:
I recommend the following string for better readability:

```perl
[ "Libkey rootkit check: The following system [numerate,library,libraries] were found which could indicate a root level compromise: [list_and,_1]", \@bad_libs ];
```

Of course, this requires an array instead of a single string of files.